custom_components/amplipi/media_player.py

```diff
@@ -1235,3 +1220,273 @@ async def async_set_volume_level(self, volume):
         if volume is None:
             return
         self._volume = volume
+
+
+class AmpliPiStream(AmpliPiMediaPlayer):
+    """Representation of an AmpliPi Stream. Supports Audio volume
+        and mute controls and the ability to change the current 'source' a
+        stream is tied to"""
+
+    def __init__(self, coordinator: AmpliPiCoordinator, namespace: str, stream: Stream,
+                 sources: List[Source],
+                 vendor: str, version: str, image_base_path: str,
+                 client: AmpliPi):
+        super().__init__(coordinator)
+        self._stream: Stream = stream
+        self._current_stream = self._stream # Make an alias for use with AmpliPiMediaPlayer functions while keeping local verbiage more correct
+        self._current_source = None
+        self._current_zones: List[Zone] = []
+        self._current_groups: List[Group] = []
+        self._sources = sources
+        self._domain = namespace
+
+        self._id = stream.id
+        self._name = stream.name
+        self._unique_id = stream.unique_id
+        self.entity_id = stream.entity_id
+        # not a real device class, but allows us to match streams and only streams with the start_streaming blueprint's streams dropdown
+        self._attr_device_class = "stream"
+        
+        self._image_base_path = image_base_path
+        self._vendor = vendor
+        self._version = version
+        self._client = client
+        self._attr_source_list = [
+            'None',
+            'Any',
+            'Source 1',
+            'Source 2',
+            'Source 3',
+            'Source 4',
+        ] if self._stream.type != "rca" else ['None', f'Source {self._stream.id - 995}']
```

Comment:
Are the special RCA id's defined anywhere? Should this use something like that instead of a magic number?